### PR TITLE
[1LP][WIPTEST] Automated Test Case for BZ 1732442

### PR DESCRIPTION
**Pull request by jawatts**
{{ pytest: cfme/tests/containers/test_container_provider_refresh.py --use-provider ocp-39-hawk }}

### CODE REVIEW

File: cfme/tests/containers/test_container_provider_refresh.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    copied_db_entry = db_entry
+
+    # Remove the object from the session
+    appliance.db.client.session.expunge(copied_db_entry)
+
+    # Make object transient
+    make_transient(copied_db_entry)
+
+    # ID is the primary key, set it to something high
+    copied_db_entry.id = 500
```
Comment (john-dupuy) [inline cfme/tests/containers/test_container_provider_refresh.py]: Could you retrieve the id of the last entry of the table and add something to it? Just to be safe

File: cfme/tests/containers/test_container_provider_refresh.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+    # Grab the first entry in the table
+    db_entry = image_query.get(1)
+
+    copied_db_entry = db_entry
+
+    # Remove the object from the session
+    appliance.db.client.session.expunge(copied_db_entry)
+
+    # Make object transient
```
Comment (john-dupuy) [inline cfme/tests/containers/test_container_provider_refresh.py]: This comment is explained by the method name, I suggest removing it or explaining what `transient` means in this context.

File: cfme/tests/containers/test_container_provider_refresh.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+            appliance.db.client.session.add(copied_db_entry)
+    except IntegrityError as ex:
+        pytest.fail(
+            'Exception while adding DB entry. {}'.format(ex)
+        )
+
+    new_db_entry = image_query.filter(image_table.id == '500').all()
+
+    # Should only be one entry
+    assert len(new_db_entry)
```
Comment (john-dupuy) [inline cfme/tests/containers/test_container_provider_refresh.py]:
```suggestion
    assert len(new_db_entry) == 1
```

File: cfme/tests/containers/test_container_provider_refresh.py [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+    for column, value in vars(new_db_entry[0]).items():
+        if column == "_sa_instance_state" or column == "id":
+            # Skip checking these
```
Comment (john-dupuy) [inline cfme/tests/containers/test_container_provider_refresh.py]: Comment is explained by the `continue`, I'd suggest removing it or explaining why you skip them.

File: cfme/tests/containers/test_container_provider_refresh.py (hunk 3 above)
Comment (jawatts) [reply cfme/tests/containers/test_container_provider_refresh.py]: Oh good catch, that is what I meant to put there

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 08aeaaf95634a60631af62cab02d9b0ee2efbc8e

The *local* fixture **`setup_temp_appliance_provider`** is used in the following files:
- **cfme/tests/containers/test_container_provider_refresh.py**
    - *test_dup_db_entry_refresh*

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

File: cfme/tests/containers/test_container_provider_refresh.py [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+    appliance = temp_appliance_preconfig
+
+    image_table = appliance.db.client['container_groups']
+
+    image_query = appliance.db.client.session.query(image_table)
+
+    all_db_entries = image_query.all()
+
+    if not all_db_entries:
+        assert False, "No Entries in the containter_groups DB table"
```
Comment (mshriver) [inline cfme/tests/containers/test_container_provider_refresh.py]: pytest.fail

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 30695b45249da5c1dd609e3849eaaca38d4c1b57

The *local* fixture **`setup_temp_appliance_provider`** is used in the following files:
- **cfme/tests/containers/test_container_provider_refresh.py**
    - *test_dup_db_entry_refresh*

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

**Pull request merged**